Large cleanup of code, spacing and comments

nahiyan8 opened the pull request:
* `RaiseToPower(x)` was removed because of the existance of the `**` operator.
* The `BITTEST(field, index)` macros can help clarity because it offers the meaning of the action at a glance.
* Please see the `FIXME` added in `code/setup.dm`, there are some re-defined macros which apparently are not used. (perhaps because the re-definition presented problems) If someone is experienced with the genetics mutation code, then please make a commit fixing that.
* `hex2num(hex)` and `num2hex(num)` speed improvements doesn't mean much.

Mloc commented on code/modules/icon generation/Uristrunes.dm:
```diff
@@ -179,7 +179,7 @@ var/list/rune_animation = list(
 
 	for(var/i = 0, i < 10, i++)
-		if(rune_bits & (1 << i))
+		if(BITTEST(rune_bits, i))
```
should be 1 << i

Mloc commented:
still not really a fan of the macros, they don't have entirely clear workings to someone who knows how to use bitwise operations already, you can't use the defined macros with them and you can't search for them in DM because of how you've formatted them.

nahiyan8 commented:
These are some details I have not mentioned in the commit messages:
* `modules/distillery.dm`: I edited some of the comments, and renamed `destilling_item` to `distilling_item`. Not much other than that, but I edited some spacing for clarity.
* `baystation12.dme`: I edited the file to include (`modules/distillery/main.dm`) accordingly with the module rename.
* `maths.dm`:
  * Rewrote `num2hex(...)` and `hex2num(...)` to be rid of the `**` operator. Independent of the other changes.
  * Removed the `var/const`s present, `E` and `Sqrt2`. (I had added `Pi` in one of my intermediate commits.) These were replaced by constants `M_E`, `M_PI` and `M_SQRT2` in `setup.dm`
  * I rewrote the `Gcd(...)` algorithm from an recursive form to an iterative form.
  * I changed `IsEven(...)` and `IsOdd(...)` to check the 1s (most insignificant) bit for oddness. I justify this by noting that the previous version used `%`, which only works on integers anyway, so this must work as well.
  * I changed `Mean(...)` to use the `.len` variable of a list, so that it doesn't have to count the number of variables it has looped over.
  * I renamed some variables in `SolveQuadratic(...)` and `Interpolate(...)`
  * I added the `squaredNorm(...)` and `norm(...)` in case anyone has to calculate a norm.
* `setup.dm`:
  * Added constants `M_E`, `M_PI` and `M_SQRT2`. The naming is based upon the C-style naming. This avoids naming collisions for `M_E` and stays concise. The instances of the old names of `PI` was changed to the new names.
  * I seperated out the `..._MIN_COLD_TEMPERATURE_PROTECTION` and `..._MAX_HEAT_PROTECTION_TEMPERATURE` macros for clarity.
  * I removed the duplicate `#define`s for `BRUTE`, `BURN`, `TOX`, etc.
  * The comment above `var/list/be_special_flags` was removed after I informed Zuhayr that it will indeed break it if the list was moved to `global.dm`.
  
That is most of it, please refer to the commit messages for additional information. The main purpose of this set of edits were to increase the clarity and style of some of the code to allow readers to understand it more easily. I have done it by using a consistent spacing style, which, however, I have not documented anywhere.

Miniature commented:
Adds spaces between if and bracket in a bunch of places inconstistently.

nahiyan8 commented:
**@Mloc:** I would like to argue that using the `BITTEST(...)` macros everywhere would let people be used to it, and I note these:
* They can be read to understand the operation at hand directly, much like when someone is experienced with bitwise operations.
* It replaces lots of tedious `if (field & (1 << index))` with `if (BITTEST(field, index))`, which is sometimes more clear when there are mulitple if's. A snippet from the code:
> if(hud_updateflag & 1 << IMPLOYAL_HUD || hud_updateflag & 1 << IMPCHEM_HUD || hud_updateflag & 1 << IMPTRACK_HUD)

Miniature commented:
Whether a thing makes that get changed in the future, currently basically everything I have seen uses no space between if and bracket (and the stuff that didn't use it was on the atmos code, which is inconsistent in a lot of other ways also), and your changes have made it inconsistent, when the apparent objective of the PR is to increase consistency. (And whether it looks cleaner is personal preference, I think it looks messier.)

comma commented:
Please do not mess with spaces and 'src' - those are just personal preference. There is little point in enforcing consitency for that over whole codebase, complete with giant diff of actually putting it so just leave them be.
As long as it's consistent with code in same file, it's fine.

nahiyan8 commented:
Closed on my request.
